Sources/Fluent/Query/QueryRepresentable.swift

```diff
@@ -173,10 +173,18 @@ extension QueryRepresentable where Self: ExecutorRepresentable {
 extension QueryRepresentable where Self: ExecutorRepresentable {
     /// Attempts to delete all entities
     /// in the model's collection.
-    public func delete() throws {
+    public func delete(shouldForceDelete: Bool = true) throws {
```

Comment:
So if I'm understanding well `delete` would always perform a soft deletion on a `SoftDeletable` object and add a method on `QueryRepresentable` when the entity is `SoftDeletable` which exposes a `forceDelete`?

I think that would be a better behaviour indeed (i.e considering the `softDelete` as default on such entities) as it seems to be how `entity.delete()` behaves.

I was initially cautious to submit a PR which would change a behaviour people may be relying on - but I guess if we consider this as a bug then we're just fixing it here.